hash.go

```diff
@@ -232,6 +235,9 @@ func (h *evpHash) sum(out []byte) {
 // The EVP_MD_CTX memory layout has changed in OpenSSL 3
 // and the property holding the internal structure is no longer md_data but algctx.
 func (h *evpHash) hashState() unsafe.Pointer {
+	if !h.marshallable {
+		panic("openssl: hash state is not marshallable")
```

Comment:
Would this ever come up from user code or OpenSSL oddity, or would there have to be a typo+missed test in golang-fips/openssl?

Wondering if it might make more sense to return this as a second value from `newEvpHash` rather than store it in the struct (at the cost of not having this runtime check).

Reply:
There would have to be a typo+missed test, aka it is a programmer error, not something that should happen at runtime.

I initially returned that property as second value from `newEvpHash`, but that made all `NewSHAX` a bit less cleaner, and having the runtime check is a nice bonus. Also, with the current implementation, `NewSHAX` functions are inlinable, which can help escape analysis take better decisions.